src/app/main/component/eco-news/components/news-preview-page/news-preview-page.component.spec.ts

```diff
@@ -0,0 +1,140 @@
+    tags: ['Events', 'Education']
+  };
+
+  const item: EcoNewsModel = {
+    author: { id: 1601, name: 'Hryshko' },
+    creationDate: '2020-10-26T16:43:29.336931Z',
+    id: 4705,
+    imagePath: 'https://storage.cloud.google.com/staging.greencity-c5a3a.appspot.com/35fce8fe-7949-48b8-bf8c-0d9a768ecb42',
+    tags: [
+      { id: 1, name: 'Events' },
+      { id: 2, name: 'Education' }
+    ],
+    text: 'hellohellohellohellohellohellohellohellohellohello',
+    title: 'hello'
+  };
+
+  const createEcoNewsServiceMock = jasmine.createSpyObj('CreateEcoNewsService', [
+    'getFormData',
+    'getNewsId',
+    'isBackToEditing',
+    'sendFormData',
+    'editNews'
+  ]);
+  createEcoNewsServiceMock.getFormData = () => currentFormWithImageMock;
+  createEcoNewsServiceMock.isBackToEditing = true;
+  createEcoNewsServiceMock.sendFormData = (previewItem) => of(newsResponseMock);
+  createEcoNewsServiceMock.editNews = (form) => of(newsResponseMock);
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [NewsPreviewPageComponent],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
+      imports: [RouterModule.forRoot([]), TranslateModule.forRoot(), HttpClientTestingModule, ReactiveFormsModule, FormsModule],
+      providers: [
+        { provide: CreateEcoNewsService, useValue: createEcoNewsServiceMock },
+        { provide: ACTION_TOKEN, useValue: ACTION_CONFIG }
+      ]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(NewsPreviewPageComponent);
+    component = fixture.componentInstance;
+    component.attributes = {
+      btnCaption: 'send',
+      title: 'title test'
+    };
+    component.images.largeImage = 'assets/img/icon/econews/news-default-large.png';
+    component.isPosting = false;
+
+    fixture.detectChanges();
+  });
+
+  it('should create component', () => {
+    createEcoNewsServiceMock.getNewsId.and.returnValue('15');
+    expect(component).toBeTruthy();
+  });
+
+  it('method isBackToEdit should change isBackToEditing to true', () => {
+    jasmine.clock().install();
+    component.isBackToEdit();
+    expect(createEcoNewsServiceMock.isBackToEditing).toBe(true);
+    jasmine.clock().tick(1001);
+    expect(!createEcoNewsServiceMock.isBackToEditing).toBe(false);
+    jasmine.clock().uninstall();
+  });
+
+  it('testing of method postNewItem', () => {
+    component.postNewsItem();
+    expect(!component.isPosting).toBe(true);
+    createEcoNewsServiceMock.sendFormData(item).subscribe(() => {
+      expect(component.isPosting).toBe(false);
+    });
+  });
+
+  it('testing of method editNews', () => {
+    component.editNews();
+    const dataToEdit = {
+      ...component.previewItem.value,
+      id: component.newsId
+    };
+    expect(dataToEdit).toBeTruthy();
+
+    createEcoNewsServiceMock.editNews(dataToEdit).subscribe(() => {
+      expect(component.isPosting).toBe(false);
+    });
+  });
```

Comment:
So, in the right way, your code should have the next structure:
`// arrange`
`const dataToEdit = ...;`
`...`
`// act`
`component.editNews();`
`...`
`// assert`
`expect(dataToEdit).toBeTruthy();`
`createEcoNewsServiceMock.editNews(dataToEdit).subscribe(() => {`
`expect(component.isPosting).toBe(false);`
`});`

So it's better to follow this structure to make your code more readable and maintainable